### PR TITLE
Refactor analysis/scheduler

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Added:
- `--quiet` option to `run analysis`

Changed:
- QoL changes to analysis, moved `scheduler.py` to `utils`, and some minor changes to `balsamic.smk`
- Removed environment variable declaration in TravisCI config

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
@ashwini06 can you please review this?

File: BALSAMIC/workflows/balsamic.smk [hunk 1]
```diff
@@ -55,23 +63,32 @@ if len(cluster_config.keys()) == 0:
 
 try:
     config["SENTIEON_LICENSE"] = os.environ["SENTIEON_LICENSE"]
-    config["SENTIEON_EXEC"] = Path(os.environ["SENTIEON_INSTALL_DIR"], "bin", "sentieon").as_posix()
+    if os.getenv("SENTIEON_EXEC") is not None:
+        config["SENTIEON_EXEC"] = os.environ["SENTIEON_EXEC"]
+    else:
+        config["SENTIEON_EXEC"] = Path(os.environ["SENTIEON_INSTALL_DIR"], "bin", "sentieon").as_posix()
+
```
Comment (ashwini06) [inline BALSAMIC/workflows/balsamic.smk:70]: :star2:

File: tests/commands/report/test_deliver.py [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
     actual_delivery_report = Path(helpers.delivery_dir,
                                   helpers.case_id + ".hk")
 
-    # WHEN running analysis
-    result = invoke_cli(
-        ['report', 'deliver', '--sample-config', tumor_only_config])
+    with mock.patch.dict(
+            'os.environ', {
+                'SENTIEON_LICENSE': sentieon_license,
+                'SENTIEON_INSTALL_DIR': sentieon_install_dir
```
Comment (ashwini06) [inline tests/commands/report/test_deliver.py:15]: :raised_hands:

**Review by ashwini06 — COMMENTED** [review]
I like the inclusion of tests for sentieon :clap: Looks good and go ahead 👍

**Pull request merged**